src/io.jl

```diff
@@ -12,31 +12,29 @@ function create_parameters_and_sets_from_file(input_folder::AbstractString)
     nodes_profiles_file = joinpath(input_folder, "nodes-profiles.csv")
     # edges_data_file     = joinpath(input_folder, "edges-data.csv")
     # edges_profiles_file = joinpath(input_folder, "edges-profiles.csv")
-    weights_file = joinpath(input_folder, "representatives-weights.csv")
+    rep_period_file = joinpath(input_folder, "rep-periods-data.csv")
 
     # Read data
     nodes_data_df     = CSV.read(nodes_data_file, DataFrames.DataFrame; header = 2)
     nodes_profiles_df = CSV.read(nodes_profiles_file, DataFrames.DataFrame; header = 2)
     # edges_data_df     = CSV.read(edges_data_file, DataFrames.DataFrame; header = 2)
     # edges_nodes_profiles_df = CSV.read(edges_profiles_file, DataFrames.DataFrame; header = 2)
-    weights_df = CSV.read(weights_file, DataFrames.DataFrame; header = 2)
+    rep_period_df = CSV.read(rep_period_file, DataFrames.DataFrame; header = 2)
 
     # Sets and subsets that depend on input data
-    A = s_assets = nodes_data_df[nodes_data_df.is_active.==true, :].node_name         #assets in the energy system that are active
-    Ap =
-        s_assets_producer = nodes_data_df[nodes_data_df.node_type.=="producer", :].node_name  #producer assets in the energy system
-    Ac =
-        s_assets_consumer = nodes_data_df[nodes_data_df.node_type.=="consumer", :].node_name  #consumer assets in the energy system
-    s_assets_investment = nodes_data_df[nodes_data_df.invest_method.=="invest", :].node_name #assets with investment method in the energy system
-    s_representative_periods = unique(nodes_profiles_df.rp)  #representative periods
-    s_time_steps = unique(nodes_profiles_df.k)   #time steps in the RP (e.g., hours)
+    A = s_assets = nodes_data_df[nodes_data_df.active.==true, :].name         #assets in the energy system that are active
+    Ap = s_assets_producer = nodes_data_df[nodes_data_df.type.=="producer", :].name  #producer assets in the energy system
+    Ac = s_assets_consumer = nodes_data_df[nodes_data_df.type.=="consumer", :].name  #consumer assets in the energy system
+    s_assets_investment = nodes_data_df[nodes_data_df.investable.==true, :].name #assets with investment method in the energy system
+    s_representative_periods = unique(nodes_profiles_df.rep_period_id)  #representative periods
+    s_time_steps = unique(nodes_profiles_df.time_step)   #time steps in the RP (e.g., hours)
 
     # Parameters for system
-    p_rp_weight = Dict((row.rp) => row.weight for row in eachrow(weights_df)) #representative period weight [h]
+    p_rp_weight = Dict((row.id) => row.weight for row in eachrow(rep_period_df)) #representative period weight [h]
```

Comment:
It's interesting to me that this shows as row.id instead of rep_period_df.id.
Does it make a difference? To me "id" is here a bit vague because it's only preceded by "row" and not "rep_period" and you have to read the whole line to see what it means.
Just a philosophical question. :P

Reply:
@abelsiqueira
@datejada and I were wondering about the philosophy behind this. Is there a more efficient/proper way?

Reply:
This is pretty efficient - maybe the most efficient.
The dictionary is created with the correct elements. The alternative would be a more traditional for loop

```julia
p_rp_weight = Dict{Int,Float64}() # The types need to be specified because they can't be inferred from the element
for row in eachrow(rep_period_df)
    p_rp_weight[row.id] = row.weight
end
```

The comprehension is short and common, and the types should be inferred, so it looks nicer.
Since "row" is the variable name, it could be changed to anything, such as, "rep_period_row". I wouldn't, for personal preference, because I think row is enough in the context, and the context is short.

That said, the "(row.id)" should be changed to "row.id", since the parenthesis does nothing.